packages/components/src/Button/Button.tsx

```diff
@@ -1,83 +1,115 @@
-import React from "react";
-import { Link } from "react-router-dom";
+import React, { forwardRef } from "react";
+import { Link, LinkProps } from "react-router-dom";
 import classnames from "classnames";
 import { ButtonProps, HTMLButtonType } from "./Button.types";
 import { useButtonStyles } from "./useButtonStyles";
 // eslint-disable-next-line import/no-deprecated
 import { ButtonContent, ButtonIcon, ButtonLabel } from "./ButtonInternals";
 import { ButtonProvider } from "./ButtonProvider";
 
-function Button(props: ButtonProps) {
-  const { size } = props;
-
-  return (
-    <ButtonProvider size={size}>
-      <ButtonWrapper {...props} />
-    </ButtonProvider>
-  );
-}
-
-function ButtonWrapper(props: ButtonProps) {
-  const {
-    ariaControls,
-    ariaHaspopup,
-    ariaExpanded,
-    ariaLabel,
-    disabled = false,
-    external,
-    id,
-    name,
-    onClick,
-    onMouseDown,
-    role,
-    value,
-    submit,
-    to,
-    url,
-    UNSAFE_className = {},
-    UNSAFE_style = {},
-    children,
-  } = props;
-
-  const { combined } = useButtonStyles(props);
-
-  const buttonType: HTMLButtonType = submit ? "submit" : "button";
-
-  const buttonClassNames = classnames(combined, UNSAFE_className.container);
-
-  const tagProps = {
-    className: buttonClassNames,
-    disabled,
-    id,
-    style: UNSAFE_style.container,
-    ...(submit && { name, value }),
-    ...(!disabled && { href: url }),
-    ...(!disabled && { onClick: onClick }),
-    ...(!disabled && { onMouseDown: onMouseDown }),
-    ...(external && { target: "_blank" }),
-    ...(url === undefined && to === undefined && { type: buttonType }),
-    "aria-controls": ariaControls,
-    "aria-haspopup": ariaHaspopup,
-    "aria-expanded": ariaExpanded,
-    "aria-label": ariaLabel,
-    role: role,
-  };
-
-  const buttonInternals = children || <ButtonContent {...props} />;
-
-  if (to) {
+// Define the possible ref types
+type ButtonRefElement = HTMLButtonElement | HTMLAnchorElement;
+
+const ButtonComponent = forwardRef<ButtonRefElement, ButtonProps>(
+  (props, ref) => {
+    const { size } = props;
+
     return (
-      <Link {...tagProps} to={to}>
-        {buttonInternals}
-      </Link>
+      <ButtonProvider size={size}>
+        <ButtonWrapper {...props} ref={ref} />
+      </ButtonProvider>
     );
-  }
+  },
+);
+ButtonComponent.displayName = "Button";
+
+const ButtonWrapper = forwardRef<ButtonRefElement, ButtonProps>(
+  // eslint-disable-next-line max-statements
```

Comment:
it was specifically upset about an arrow function

could try changing this to a named function and perhaps it will be happy, sparing us from disabling eslint